File: features/login/presentation/src/androidDeviceTest/kotlin/com/mockdonalds/app/features/login/presentation/LoginStateRobot.kt
package com.mockdonalds.app.features.login.presentation

import com.mockdonalds.app.core.test.StateRobot

class LoginStateRobot : StateRobot<LoginUiState, LoginEvent>() {

    override fun defaultState() = LoginUiState(
        logoUrl = "",
        email = "",
        password = "",
        isLoading = false,
        errorMessage = null,
        eventSink = createEventSink(),
    )

    fun stateWithEmail(email: String) = defaultState().copy(
        email = email,
        eventSink = createEventSink(),
    )

    fun stateWithCredentials(email: String, password: String) = defaultState().copy(
        email = email,
        password = password,
        eventSink = createEventSink(),
    )
}

File: iosApp/iosAppTests/Login/LoginStateRobot.swift
import ComposeApp

final class LoginStateRobot: BaseStateRobot<LoginUiState, LoginEvent> {

    override func defaultState() -> LoginUiState {
        LoginUiState(
            logoUrl: "",
            email: "",
            isLoading: false,
            errorMessage: nil,
            eventSink: createEventSink()
        )
    }

    func stateWithEmail(_ email: String) -> LoginUiState {
        LoginUiState(
            logoUrl: "",
            email: email,
            isLoading: false,
            errorMessage: nil,
            eventSink: createEventSink()
        )
    }
}
